pkg/handlers/defaulthandler.go

```diff
@@ -144,22 +196,28 @@ func (h *defaultHandler) extractorHandler(archiveChan chan []byte) func(context.
 		if ctxDepth, ok := ctx.Value(depthKey).(int); ok {
 			depth = ctxDepth
 		}
-		if int(file.Size()) > maxSize {
+
+		fileSize := file.Size()
+		if int(fileSize) > maxSize {
 			lCtx.Logger().V(3).Info("skipping file due to size")
 			return nil
 		}
 
 		if common.SkipFile(file.Name()) {
 			lCtx.Logger().V(5).Info("skipping file")
+			h.metrics.incFilesSkipped()
 			return nil
 		}
 
 		fReader, err := file.Open()
 		if err != nil {
-			return err
+			return fmt.Errorf("error opening file %s: %w", file.Name(), err)
```

Comment:
any reason to not use `%q` instead of `%s`? i find it can highlight weird file names better.

Reply:
I always forget about `%q`